tools/clang/test/CodeGenHLSL/expressions/conversions_and_casts/between_type_shapes.hlsl

```diff
@@ -0,0 +1,442 @@
+    // DXC crashes (GitHub #1799)
+    // FXC: l(1,0,0,0)
+    // output_m1x1((int1x1)s1);
+
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_a1((A1)i);
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_a1((A1)v1);
+    // DXC crashes (GitHub #1799)
+    // FXC: l(11,0,0,0)
+    // output_a1((A1)m1x1);
+    // DXC rejects (GitHub #1862)
+    // FXC: l(1,0,0,0)
+    // output_a1(s1); 
+    // DXC crashes (GitHub #1799)
+    // FXC: l(1,0,0,0)
+    // output_a1((A1)s1);
+
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_s1((S1)i);
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_s1((S1)v1);
+    // DXC crashes (GitHub #1799)
+    // FXC: l(11,0,0,0)
+    // output_s1((S1)m1x1);
+    // DXC rejects (GitHub #1862)
+    // FXC: l(1,0,0,0)
+    // output_s1(a1);
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_s1((S1)a1);
+
+    output_separator();
+
+    // =========== Truncation to scalar/single-element ===========
+    // Single element sources already tested
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_i(v2); // warning: implicit truncation of vector type
+    // DXC: i32 11, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(11,0,0,0)
+    output_i(m2x2); // warning: implicit truncation of vector type
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_i((int)a2);
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_i((int)s2);
+
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_v1(v2); // warning: implicit truncation of vector type
+    // DXC: i32 11, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(11,0,0,0)
+    output_v1(m2x2); // warning: implicit truncation of vector type
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_v1((int1)a2);
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_v1((int1)s2);
+
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_m1x1(v2); // warning: implicit truncation of vector type
+    // DXC: i32 11, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(11,0,0,0)
+    output_m1x1(m2x2); // warning: implicit truncation of vector type
+    // DXC crashes (GitHub #1799)
+    // FXC: l(1,0,0,0)
+    // output_m1x1((int1x1)a2);
+    // DXC crashes (GitHub #1799)
+    // FXC: l(1,0,0,0)
+    // output_m1x1((int1x1)s2);
+
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_a1((A1)a2);
+    // DXC crashes (GitHub #1799)
+    // FXC: l(1,0,0,0)
+    // output_a1((A1)s2);
+
+    // DXC: i32 1, i32 0, i32 0, i32 0, i8 15)
+    // FXC: l(1,0,0,0)
+    output_s1((S1)a2);
+    // DXC crashes (GitHub #1799)
+    // FXC: l(1,0,0,0)
+    // output_s1((S1)s2);
+    
+    output_separator();
```

Comment:
It might be useful to add `// DXC: i32 888, i32 888, i32 888, i32 888, i8 15)` for better match analysis in case there are failures.

Reply:
Good idea! Tracking down the line causing a break has been painful before. I was thinking I might split the test into multiple files, but this is simpler.

Reply:
I've followed your suggestion and will do the splitting into multiple files in a different change.